petri/src/vm/openvmm/runtime.rs

```diff
@@ -166,6 +166,11 @@ impl PetriVmOpenVmm {
         /// method is best effort for them.
         pub async fn wait_for_successful_boot_event(&mut self) -> anyhow::Result<()>
     );
+    petri_vm_fn!(
+        /// Waits for the Hyper-V shutdown IC to be ready, returning a receiver
+        /// that will be closed when it is no longer ready.
```

Comment:
This feels like an odd contract: a receiver that will never receive, only close? I guess it makes sense, but is there no better way to model this?

Reply:
I have a comment that didn't make it into the PR that will clarify this. Just a sec.